src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+        }
+        voteWeigher.modifyStrategyWeights(quorumNumber, strategyIndices, newWeights);
+
+        // convert the strategies and weighting multipliers to the modified
+        for (uint i = 0; i < strategyIndices.length; i++) {
+            strategiesAndWeightingMultipliers[strategyIndices[i]].multiplier = newWeights[i];
+        }
+        // make sure the quorum strategies and weights have changed
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            IVoteWeigher.StrategyAndWeightingMultiplier memory strategyAndWeightingMultiplier = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumNumber, i);
+            assertEq(address(strategyAndWeightingMultiplier.strategy), address(strategiesAndWeightingMultipliers[i].strategy));
+            assertEq(strategyAndWeightingMultiplier.multiplier, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+    }
+
+    function testModifyStrategyWeights_ForNonExistantQuorum_Reverts(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory strategyIndices,
+        uint96[] memory newWeights
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+
+        // create a valid quorum
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.startPrank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+
+        // modify certain strategies of a non-existant quorum
+        cheats.expectRevert("VoteWeigherBase.modifyStrategyWeights: quorumNumber is greater than or equal to quorumCount");
+        voteWeigher.modifyStrategyWeights(quorumNumber + 1, strategyIndices, newWeights);
+    }
+
+    function testModifyStrategyWeights_InconsistentStrategyAndWeightArrayLengths_Reverts(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers,
+        uint256[] memory strategyIndices,
+        uint96[] memory newWeights
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+
+        // make sure the arrays are of different lengths
+        cheats.assume(strategyIndices.length != newWeights.length);
+
+        // create a valid quorum
+        uint8 quorumNumber = voteWeigher.quorumCount();
+        cheats.startPrank(serviceManagerOwner);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+
+        // modify certain strategies
+        cheats.expectRevert("VoteWeigherBase.modifyStrategyWeights: input length mismatch");
+        voteWeigher.modifyStrategyWeights(quorumNumber, strategyIndices, newWeights);
+    }
+
+    function testModifyStrategyWeights_EmptyStrategyIndicesAndWeights_HasNoEffect(
```

Comment:
I again think this is a case where reverting probably makes more sense than having no effect. current behavior seems acceptable though and this test looks alright.

Reply:
agreeed, fixed